program/src/reward_centers/create/mod.rs

```diff
@@ -26,14 +26,19 @@ pub struct CreateRewardCenter<'info> {
     #[
       account(
         mut,
-        constraint = wallet.key() == auction_house.authority @ ListingRewardsError::SignerNotAuthorized
+        constraint = wallet.key() == auction_house.authority @ RewardCenterError::SignerNotAuthorized
       )
     ]
     pub wallet: Signer<'info>,
 
     /// the mint of the token to use as rewards.
+    #[account(constraint = mint.decimals == auction_house_treasury_mint.decimals @ RewardCenterError::RewardMintDecimalMismatch)]
     pub mint: Account<'info, Mint>,
 
+    // the mint of the accepted token currency for the associated auction house
+    #[account(constraint = auction_house.treasury_mint.key() == auction_house_treasury_mint.key() @ RewardCenterError::AuctionHouseTreasuryMismatch)]
+    pub auction_house_treasury_mint: Account<'info, Mint>,
+
```

Comment:
Pass the treasury mint of the auction house as an account so can match its decimals with the reward mint. They need to match for clean mass with no remainders.